index.js

```diff
@@ -6,17 +6,19 @@
 var CSSescape = require('css.escape')
 
 exports.hash = function hash (h, options) {
-  if (!h) return
   options = options || {}
+  var cb = options.callback || function () {}
+  // There's no hash to scroll to, so "success", we did it.
+  if (!h) return void cb(false)
   var scroller = options.scroller || exports.scroller
   // Push onto callback queue so it runs after the DOM is updated,
   // this is required when navigating from a different page so that
   // the element is rendered on the page before trying to getElementById.
   setTimeout(function () {
     var els = exports.elements(h)
-    if (!els) return
-    if (scroller(els.id)) return
-    scroller(els.name)
+    if (!els) return void cb(true)
+    if (scroller(els.id)) return void cb(false)
+    cb(!scroller(els.name))
```

Comment:
Ditto https://github.com/AndersDJohnson/anchorate/pull/2/files#r168876130, I would prefer just `cb()` here - passing `undefined`. So maybe refactor to:
```js
if (scroller(els.id)) return void cb()
if (scroller(els.name)) return void cb()
cb(true)
```

Reply:
As style owner would you prefer that, or something like:
```js
if (scroller(els.id) || scroller(els.name)) return void cb()
cb(true)
```

Reply:
Went with your initial suggestion for now